python_modules/dagster/dagster/_core/loader.py

```diff
@@ -58,15 +58,19 @@ def instance(self) -> "DagsterInstance":
     def loaders(self) -> Dict[Type, Tuple[DataLoader, BlockingDataLoader]]:
         raise NotImplementedError()
 
+    @staticmethod
+    def ephemeral(instance: "DagsterInstance") -> "LoadingContext":
+        return EphemeralLoadingContext(instance)
+
     def get_loaders_for(
         self, ttype: Type["InstanceLoadableBy"]
     ) -> Tuple[DataLoader, BlockingDataLoader]:
         if ttype not in self.loaders:
             if not issubclass(ttype, InstanceLoadableBy):
                 check.failed(f"{ttype} is not Loadable")
 
-            batch_load_fn = partial(ttype._batch_load, instance=self.instance)  # noqa
-            blocking_batch_load_fn = partial(ttype._blocking_batch_load, instance=self.instance)  # noqa
```

Comment:
when I first set this up I imagined we might have other `_X_LoadableBy`  which would bind `_X_` instead of the `instance` for fetching, so in this case `ContextLoadableBy` 

i'm not sure that having different flavors is necessary in the end here - but we might want to drop the `Instance` part of the naming for LoadableBy interface if we just want to always pass the context

Reply:
Yeah my thinking was that in a world where there are multiple things that we want to load with (i.e. multiple Xs), there would likely conceivably be singular objects that would want more than one of those Xs, and so at that point we'd probably just do this change anyway.

Agree `LoadableBy` would be a better name (maybe that could be a followup to this PR?)

Reply:
follow up or in this PR is fine with me